api/src/main/java/org/openmrs/module/reporting/cohort/definition/library/BuiltInCohortDefinitionLibrary.java

```diff
@@ -171,4 +172,12 @@ public CohortDefinition getPatientsWithEnrollment(){
             "locationList", "locations"
         );
     }
+
+    @DocumentedDefinition("encounterLocationSearch")
```

Comment:
This should not be needed.  We should just be able to use encounterSearchAdvanced for this (adding the TimeQualifier as an optional property).  Now that I notice it, we should also change the method name of the encounterSearchAdvanced from "getAnyEncounterDuringPeriodWithOccurrence" to "getEncounterSearchAdvanced"